### PR TITLE
feat: direct upload with skip verify: set metadata on complete

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-6696

### CODE REVIEW

**Comment by merlin-northern** [comment]
friendly mender-cli PR:https://github.com/mendersoftware/mender-cli/pull/220

File: api/http/api_deployments.go [hunk 1]
```diff
@@ -427,13 +428,34 @@ func (d *DeploymentsApiHandlers) UploadLink(w rest.ResponseWriter, r *rest.Reque
...
+	var metadata *model.DirectUploadMetadata
+	if d.config.EnableDirectUploadSkipVerify {
+		var directMetadata model.DirectUploadMetadata
+		bodyBuffer := make([]byte, maxMetadataSize)
+		n, err := io.ReadFull(r.Body, bodyBuffer)
+		r.Body.Close()
+		if err != nil && err != io.EOF && err != io.ErrUnexpectedEOF {
+			l.Errorf("error reading post body data: %s (read: %d)", err.Error(), n)
+		}
+		err = json.Unmarshal(bodyBuffer[:n], &directMetadata)
```
Comment (tranchitella) [inline api/http/api_deployments.go:448]: If `err != nil`, you don't want to continue with the Unmarshal, right?
Maybe you should stuff this code in an `else`.

**Review by tranchitella — COMMENTED** [review]
The general structure looks good!

File: api/http/api_deployments.go (hunk 1 above)
Comment (merlin-northern) [reply api/http/api_deployments.go:448]: fixed.

**Review by tranchitella — APPROVED** [review]

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline --fast please

**Pull request merged**